app/javascript/dashboard/components-next/NewConversation/components/ActionButtons.vue

```diff
@@ -77,18 +84,30 @@ const toggleMessageSignature = () => {
   setSignature();
 };
 
-// Added this watch to dynamically set signature.
 // Only targetInbox has value and is Advance Editor(used by isEmailOrWebWidgetInbox)
 // Set the signature only if the inbox based flag is true
+const handleSignatureSetup = () => {
+  nextTick(() => {
+    if (props.hasSelectedInbox && props.isEmailOrWebWidgetInbox) {
+      setSignature();
+    }
+  });
+};
+
+// Added this watch to dynamically set signature on target inbox change.
 watch(
   () => props.hasSelectedInbox,
   newValue => {
-    nextTick(() => {
-      if (newValue && props.isEmailOrWebWidgetInbox) setSignature();
-    });
+    if (newValue) handleSignatureSetup();
   }
 );
```

Comment:
Instead of `mounted`, can we add an `immediate: true` here?

Reply:
@scmmishra Done